graphics/blit-atari.cpp

```diff
@@ -37,27 +38,46 @@ void Surface::create(int16 width, int16 height, const PixelFormat &f) {
 	assert(width >= 0 && height >= 0);
 	free();
 
+	constexpr size_t ALIGN = 16;	// 16 bytes
+
 	w = width;
 	h = height;
 	format = f;
-	pitch = w * format.bytesPerPixel;
+	// align pitch to a 16-byte boundary for a possible C2P optimization
+	pitch = (w * format.bytesPerPixel + ALIGN - 1) & (-ALIGN);
 
 	if (width && height) {
-		if (VgetMonitor() == MON_VGA && Getcookie(C_SupV, NULL) == C_FOUND)
-			pixels = (void*)ct60_vmalloc(width * height * format.bytesPerPixel);
-		else
-			pixels = calloc(width * height, format.bytesPerPixel);
-		assert(pixels);
+		if (VgetMonitor() == MON_VGA && Getcookie(C_SupV, NULL) == C_FOUND) {
+			pixels = (void *)ct60_vmalloc(height * pitch);
+
+			if (!pixels)
+				error("Not enough SVRAM to allocate a surface");
+
+			assert((uintptr)pixels >= 0xA0000000);
+		} else {
+			// align buffer to a 16-byte boundary for move16
+			void *pixelsUnaligned = ::malloc(sizeof(uintptr) + (height * pitch) + ALIGN - 1);
+
+			if (!pixelsUnaligned)
+				error("Not enough memory to allocate a surface");
+
+			pixels = (void *)(((uintptr)pixelsUnaligned + sizeof(uintptr) + ALIGN - 1) & (-ALIGN));
+
+			// store the unaligned pointer for later free()
+			*((uintptr *)pixels - 1) = (uintptr)pixelsUnaligned;
+		}
+
+		memset(pixels, 0, height * pitch);
 	}
 }
 
 void Surface::free() {
 	if (((uintptr)pixels & 0xFF000000) >= 0xA0000000)
 		ct60_vmfree(pixels);
-	else
-		::free(pixels);
+	else if (pixels)
+		::free((void *)*((uintptr *)pixels - 1));
```

Comment:
This looks suspicious, as I did not see you incrementing pixels pointer by 1. But of course, you know what is going on on your OS